### PR TITLE
add

### PR DESCRIPTION
**Pull request by LiuZesensengsheng**
这里运行python test.py 之后  会卡在discrete_plan.py  中  function： compute_path_from_pre 的死循环中。  暂时不知道是这里错了还是之前就发生错误了。还在尝试和Python2下的程序进行对比数据。

### CODE REVIEW

**Comment by LiuZesensengsheng** [comment]
I have a good news and a bad news:
The good news is that the program can work quickly and the value of precost and sufcost can be quite similar to the old one.
The bad news is the prefix and and sufix are different from the old one ! I hope the it is because the new version of networkx function dijkstra_plan_networkX are changed and it seems to work as good as the old one throught the different prefix and sufix. X(
I confuse that is the program doesn't work worse or the the program answer incorrent worse?

**Comment by MengGuo** [comment]
你现在还是在测试一个智能体的情况是吧？
嗯，产生的plan不一样很正常，一般是因为有很多相同的path.

几个建议：
* 把`__pycache__`放到`.gitignore`里面
* 重新把code过一遍，现在不用的function都去掉，比如[DuoBA_from_ltls](https://github.com/LiuZesensengsheng/LTL_MAS_C-action/blob/main/LTL_MAS_TS/buchi.py#L34) 和on-the-fly construction(https://github.com/LiuZesensengsheng/LTL_MAS_C-action/blob/main/LTL_MAS_TS/ts.py#L185-L230)
* 你在做的测试可以考虑用[unit test](https://www.datacamp.com/community/tutorials/unit-testing-python). 这样就可以自动测试了。
* 重组结构，见下面。
* 早期实现visualization，更直观。

在编多个智能体的时候要注意框架，我建议大致上：
```shell
models/ （里面放结构定义和框架，很重要，一般不含算法）
        dynamic_model.py
        buchi.py
        ts.py
        product.py
        agent.py （智能体class就包含ts,buchi,product,agent_planner）
        agent_planner.py
        system.py (系统class就包含多个agent,和ltl task,和协作任务task)
        system_planner.py
        task.py （可以用来定义协作任务）

 
tools/ （里面放算法）
        promela.py
        ltl2ba.py
        agent_discrete_plan_algorithms.py
        system_discrete_plan_algorithms.py
        agent_motion_control.py
        boolean_formulas/
                lexer.py
                parser.py

visualization/
        plot_....
```
你觉得怎么样？其实你把code结构弄清楚，写paper也清晰:laughing:

**Review by MengGuo — CHANGES_REQUESTED** [review]

File: LTL_MAS_TS/discrete_plan.py [hunk 1]
```diff
@@ -179,6 +179,12 @@ def compute_path_from_pre(pre, target):#here occur the dead circle!!
 	#print('n==',n)
 	#print('pre=',pre)
 	while (n in pre):
+		'''
```
[the remark this answers is not in the document]
Comment (LiuZesensengsheng) [reply LTL_MAS_TS/discrete_plan.py:182]: OOOK! My plan of next week is to build the collaboration-action and the visualization.

[the remark this answers is not in the document]
Comment (MengGuo) [reply LTL_MAS_TS/discrete_plan.py:182]: Sounds good. You may send a PR for the new structure first.

**Pull request merged**